File: GFG/LinkedList/revk.cpp
#include<iostream>
#include"../ds.h"
using namespace std;


Node<int>* revk(Node<int>* head,int k){
    Node<int>* curr=head;
    Node<int>* prev=NULL,*nert=NULL;
    int i=0;
    while(curr!=NULL && i<k){
        nert=curr->next;
        curr->next=prev;
        prev=curr,curr=nert;
        i++;
    }
    if(nert!=NULL){
        Node<int>* rest_head=revk(nert,k);
        head->next=rest_head;
    }

    return prev;
}

int main(){
    LinkedList<int> dll;
    for(int i=1;i<21;i++){
        dll.insert(i*10);
    }
    dll.display();
    dll.head=revk(dll.head,3);
    cout<<endl;
    dll.display();
    return 0;
}

File: GFG/LinkedList/revk.py
import ds

def revk(head,k):
    curr=head
    nert=None
    prev=None
    count=0
    while(curr!=None and count<k):
        nert=curr.next
        curr.next=prev
        prev=curr
        curr=nert
        count+=1
    if(nert!=None):
        rest_head=revk(nert,k)
        head.next=rest_head
    return prev

dll=ds.LinkedList()
for i in range(1,11):
    dll.insert(i*10)
dll.display()
print("")
dll.head=revk(dll.head,3)
dll.display()
        
    